src/libs/stripe-lib.js

```diff
@@ -115,6 +115,23 @@ class StripeLib {
       throw new PaymentError(error.message, error.statusCode);
     }
   }
+
+  /**
+   * @param { Object } payload
+   * @param { number } payload.limit
+   * @param { string } payload.startingAfter
+   */
+  async getInvoices(payload) {
+    try {
+      const { limit, startingAfter } = payload;
+      return await this.stripe.invoices.list({
```

Comment:
I think the flow should go from orders

I.e. we fetch orders that were created <= date.now and have status !=='paid'

then get its invoice from stripe and update order, instead of listing all invoices from stripe and loop over them

also when fetching orders from our db, we can loop over them with Promise.map and give some concurrency, to not load all orders into memory

Promise.map(orders, asyncFn, { concurrency: 5 });